### DISCUSSION TITLE
Shouldn't handleSubmit return () => void?

### DISCUSSION THREAD

**Discussion by fakkio** [react-hook-form/react-hook-form#10819] [Q&A]
Since the form tag's onSubmit attribute is of type `FormEventHandler<HTMLFormElement> = (e: FormEvent<HTMLFormElement>) => void`, shouldn't `handleSubmit()` also return that type instead of `(e: ...) =>Promise<void>`?

Is there a reason why the return type is like this?

Fabio.

**Comment by Moshyfawn** [comment]
`handleSubmit` manages various submission states, including the `isSubmitting` state. When the submission operation is asynchronous and represented as a promise, the `isSubmitting` state is set to true until the promise resolves; Same goes for the successful / erroneous states.

**Comment by fakkio** [comment]
Sure, I see, but the promise can be ran, resolved or rejected even without it being returned by the function, isn't it?
I don't understand what advantage there is in returning a promise which will then be completely ignored by the form tag and which then creates problems with eslint's stricter rules.
